### PR TITLE
feat: allow init_args to point to contract address

### PR DESCRIPTION
**Pull request by Alejandro-Morales**
These updates allow contracts to be set in a specific order of deployment to be able to use deployed contract addresses in the instantiation of other contracts.

### CODE REVIEW

File: src/jenesis/contracts/deploy.py [hunk 1]
```diff
@@ -16,6 +19,16 @@
 from jenesis.tasks import Task, TaskStatus
 from jenesis.tasks.monitor import run_tasks
 import os
+import toml
+import graphlib as gl
+
+# Recursive function to insert deployed contract address into instantiation msg
```
Comment (jrriehl) [inline src/jenesis/contracts/deploy.py:25]: This is nice!

File: src/jenesis/contracts/deploy.py [hunk 2]
```diff
@@ -154,72 +164,105 @@ def deploy_contracts(cfg: Config, project_path: str, deployer_key: Optional[str]
...
-            contracts_to_deploy.append((contract, profile_contract))
-            continue
+        # determine what tasks to do
+        for contract in contracts:
 
-        digest = contract.digest()
-        if digest is None:
-            continue  # we can't process any contracts where we don't have
+            if contract.name != C:
+                continue
```
Comment (jrriehl) [inline src/jenesis/contracts/deploy.py:184]: This seems somewhat inefficient. Maybe better to set `contracts = {contract.name: contract for contract in detect_contracts(...)}` for easier indexing by name?

Comment (Alejandro-Morales) [reply src/jenesis/contracts/deploy.py:184]: Yes, you are right. Ill take a look

File: src/jenesis/contracts/deploy.py [hunk 3]
```diff
@@ -153,13 +197,25 @@ def deploy_contracts(cfg: Config, project_path: str, deployer_key: Optional[str]
...
-    # determine what tasks to do
-    for contract in contracts:
+    init_addresses = {
+        name: set(profile_contracts[name]["init_addresses"])
+        for name in contracts_list.keys()
+    }
+
+    sorter = gl.TopologicalSorter(init_addresses)
+    deployment_order = list(sorter.static_order())
+    contract_to_deploy = ""
```
Comment (jrriehl) [inline src/jenesis/contracts/deploy.py:212]: It's not clear to me why `contract_to_deploy` is necessary in the loop below. I also would've expected it to get reset to `""` at the beginning of each loop iteration.

Would it be a bit more straightforward to simply `continue` the loop whenever one of the conditions (configuration not out of date or digest not changed) holds?

Comment (Alejandro-Morales) [reply src/jenesis/contracts/deploy.py:212]: Yeah that is a good idea! Ill get to it

**Pull request merged**